prow/clonerefs/run.go

```diff
@@ -36,7 +37,57 @@ import (
 	"k8s.io/test-infra/prow/pod-utils/clone"
 )
 
+// Files to be mounted before cloning, this SSH fingerprint is a well known
+// GitHub fingerprint.
+// ref: https://github.com/kubernetes/test-infra/issues/9638
+func mountDefaultGithubFingerprint(dir string) error {
+	var sshConfigFiles = map[string]string{
+		"/etc/ssh/ssh_config": "github.com ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEAq2A7hRGmdnm9tUDbO9IDSwBK6TbQa+PXYPCPy6rbTrTtw7PHkccKrpp0yVhp5HdEIcKr6pLlVDBfOLX9QUsyCOV0wzfjIJNlGEYsdlLJizHhbn2mUjvSAHQqZETYP81eFzLQNnPHt4EVVUh7VfDESU84KezmD5QlWpXLmvU31/yMf+Se8xhHTvKSCZIFImWwoG6mbUoWf9nzpIoaSjB+weqqUUmpaaasXVal72J+UX2B+2RPW3RcT0eOzQgqlJL3RKrTJvdsjE3JEAvGq3lGHSZXy28G3skua2SmVi/w4yCE6gbODqnTWlg7+wC604ydGXA8VJiS5ap43JXiUFFAaQ==",
+		"/github_known_hosts": `Host github.com
+	HostName github.com
+	User git
+	UserKnownHostsFile /github_known_hosts
+	StrictHostKeyChecking yes
+	CheckHostIP no`,
+	}
+
+	// Best effort mounting ssh_config and know_hosts file
+	for fp, content := range sshConfigFiles {
+		// Normalize fp
+		fp = path.Join(dir, fp)
+		// Not to add it repeatedly
+		if existing, err := ioutil.ReadFile(fp); err == nil && strings.Contains(string(existing), content) {
+			continue
+		}
+
+		// Create dir if not exist
+		dir := path.Dir(fp)
+		if _, err := os.Lstat(dir); err != nil {
+			if !os.IsNotExist(err) {
+				return fmt.Errorf("stats dir '%s': %w", dir, err)
+			}
+			if err := os.MkdirAll(dir, 0777); err != nil {
+				return fmt.Errorf("create dir '%s': %w", dir, err)
+			}
+		}
+		// If the file doesn't exist, create it, or append to the file
+		f, err := os.OpenFile(fp, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+		if err != nil {
+			return fmt.Errorf("open file '%s': %w", fp, err)
+		}
+		if _, err := f.Write([]byte(content + "\n")); err != nil {
+			f.Close() // ignore error; Write error takes precedence
```

Comment:
Might as well log it, just so it's visible

Reply:
done